src/main/kotlin/eu/long1/flutter/i18n/workers/I18nFileGenerator.kt

```diff
@@ -341,11 +337,23 @@ import 'package:flutter/material.dart';
 """
 
         private const val sClassHeader =
-            """class S implements WidgetsLocalizations {
+            """typedef void LocaleChangeCallback(Locale locale);
+
+class S implements WidgetsLocalizations {
   const S();
 
   static S current;
 
+  static Locale _locale;
+  static bool _shouldReload = false;
```

Comment:
This variable is redundant, you can detect a new Locale in `bool shouldReload(GeneratedLocalizationsDelegate)` and return the result there.

Reply:
I don't see how I could detect the new locale in the method.
Unless I add new attributes to the class `GeneratedLocalizationsDelegate` to hold current locale.
The `S` class holds the logic of changing the locale.